File: BE-FindingJob/src/main/java/com/example/befindingjob/entity/Skill.java
package com.example.befindingjob.entity;

import com.fasterxml.jackson.annotation.JsonBackReference;
import com.fasterxml.jackson.annotation.JsonIgnore;
import com.fasterxml.jackson.annotation.JsonManagedReference;
import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import java.util.HashSet;
import java.util.Set;

@Entity
@Table(name = "skills")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
public class Skill {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer skillId;

    @Column(nullable = false, unique = true)
    private String skillName;

    @JsonIgnore
    @ManyToMany(mappedBy = "skills")
    private Set<User> users = new HashSet<>();

    @JsonIgnore
    @ManyToMany(mappedBy = "requiredSkills")
    private Set<Job> jobs = new HashSet<>();
}

File: JobHub/app/src/main/java/com/example/jobhub/entity/Skill.kt
package com.example.jobhub.entity

import android.os.Parcelable
import kotlinx.parcelize.Parcelize

@Parcelize
data class Skill(
    var skillId: Int? = null,
    var skillName: String = "",
    var users: Set<User>? = emptySet(),
    var jobs: Set<Job> = emptySet()
) : Parcelable

